### ISSUE TITLE
pscheduler: Unable to determine the lead participant

### ISSUE THREAD

**Issue by igarny** [perfsonar/pscheduler#427]
[Edited for readability.  --Mark]
Hi Mark,

from a task specification like this:
```
pscheduler task
  --assist ps-owdtst.rrze.uni-erlangen.de
  --bind test-rhps02.geant.net
  --debug
throughput
  --dest-node test-rhps02.geant.net
  --dest 62.40.123.106
  --source-node ps-owdtst.rrze.uni-erlangen.de
  --source psps-test2-mgmt.rrze.uni-erlangen.de
```
 where I would like to request a test from the Remote party -> Local host
I see pscheduler is correctly connecting to the Remote with a complete task specification. Nonetheless pscheduler is IMHO disregarding the `--source-node` param (which is its local address) and is not able to comprehend the Lead participant.

```
pscheduler task --assist ps-owdtst.rrze.uni-erlangen.de --bind test-rhps02.geant.net --debug throughput --dest-node test-rhps02.geant.net --dest 62.40.123.106  --source-node ps-owdtst.rrze.uni-erlangen.de --source psps-test2-mgmt.rrze.uni-erlangen.de
2017-05-05T09:03:07 Debug signal ignored; already not debugging
2017-05-05T09:03:07 Debug discontinued
2017-05-05T09:03:07 Assistance is from ps-owdtst.rrze.uni-erlangen.de
2017-05-05T09:03:07 Using slip of PT5M
2017-05-05T09:03:08 Converting to spec via https://ps-owdtst.rrze.uni-erlangen.de/pscheduler/tests/throughput/spec
Submitting with assistance from ps-owdtst.rrze.uni-erlangen.de...
2017-05-05T09:03:16 Fetching participant list
2017-05-05T09:03:16 Spec is: {"dest": "62.40.123.106", "source": "psps-test2-mgmt.rrze.uni-erlangen.de", "source-node": "ps-owdtst.rrze.uni-erlangen.de", "dest-node": "test-rhps02.geant.net", "schema": 1}
Unable to determine the lead participant: Can't find pScheduler or BWCTL on test-rhps02.geant.net
```

**Comment by igarny** [comment]
of course adding a --lead-bind param resolves it (please check for my other bug report)
still I am looking for an optimization of the params

**Comment by mfeit-internet2** [comment]
It isn't being ignored.  The `participants` method in the throughput test plugin picks out  and uses `source-node` if it's there, so it isn't being ignored:  https://github.com/perfsonar/pscheduler/blob/master/pscheduler-test-throughput/throughput/participants#L20

The cause of it not working properly is the BWCTL backward compatibility code.  There isn't a decent way to make it work as you want it without writing a lot of code.  The problem will go completely away in 4.1, when the BC hack is removed.

**Comment by mfeit-internet2** [comment]
Closing; the code that fixes this has already been committed to 4.2.0.
